app/models/content.rb

```diff
@@ -11,11 +12,58 @@ def validate_episode_medium
     end
   end
 
+  def validate_segmentation
+    return if segmentation.nil?
+
+    # can be [1.23, 4.56] or [nil, 4.56] or [1.23, nil]
+    unless array_segments? && numeric_segments? && ordered_segments?
+      errors.add(:segmentation, :bad_slices, message: "bad slices")
+    end
+  end
+
+  def slice?
+    segmentation.present?
+  end
+
+  def slice_start
+    segmentation.try(:[], 0)
+  end
+
+  def slice_start=(time)
+    self.segmentation = [time.try(:zero?) ? nil : time, slice_end]
+  end
+
+  def slice_end
+    segmentation.try(:[], 1)
+  end
+
+  def slice_end=(time)
+    self.segmentation = [slice_start, time]
+  end
+
   def publish_episode!
     episode&.publish! if status_complete? && status_previously_changed?
   end
 
+  def replace?(res)
+    super || segmentation != res.try(:segmentation)
```

Comment:
Using the usual `episode.media=` setter to create/replace these sliced segments.  Internally, it calls this method to decide if we should replace or just update.